components/cmake/cmake_util.cmake

```diff
@@ -66,10 +66,22 @@ function(e3sm_remove_flags FILE_ARG FLAGS_ARG)
     if (NOT EXISTS ${PROJECT_SOURCE_DIR}/${FILE_ARG})
       message(FATAL_ERROR "Trying to set flags on non-existent source: ${FILE_ARG}")
     endif()
-    set(REAL_FILE "../../${FILE_ARG}")
+    set(REAL_FILE "${SOURCE_PATH}/${FILE_ARG}")
   endif()
 
   get_property(ITEM_FLAGS SOURCE ${REAL_FILE} PROPERTY COMPILE_FLAGS)
   string(REPLACE "${FLAGS_ARG}" "" ITEM_FLAGS "${ITEM_FLAGS}")
   set_property(SOURCE ${REAL_FILE} PROPERTY COMPILE_FLAGS "${ITEM_FLAGS}")
 endfunction()
+
+function(e3sm_deoptimize_file FILE_ARG FFLAGS_NOOPT)
+  if (CMAKE_Fortran_COMPILER_ID STREQUAL "PGI")
```

Comment:
So, you are removing existing flags applies to PGI only as that's the troublemaker, correct? Rest of the compiler families would keep appending flags. Just something we should record for future reference.

Reply:
Yes, see comment below.